core/metacling/src/TClingCallFunc.cxx

```diff
@@ -917,7 +856,7 @@ int TClingCallFunc::get_wrapper_code(std::string &wrapper_name, std::string &wra
       } break;
       }
    }
-   unsigned min_args = GetMinRequiredArguments();
```

Comment:
Calculating the `getMinRequiredArguments()` is a costly operation; we saw it bubble up in several perf profiles. That's why we cache it. Is this not an issue anymore?

Reply:
Fair point, I thought about that when making this change. I think it will mostly be relevant in the `::exec` call. I will revert that change. However, if `::exec` is the hotspot we do whole bunch of operations there which can be cached as well...